test/integration/scheduler/preemption_test.go

```diff
@@ -490,6 +491,119 @@ func TestPreemptionStarvation(t *testing.T) {
 	}
 }
 
+// TestPreemptionRaces tests that other scheduling events and operations do not
+// race with the preemption process.
+func TestPreemptionRaces(t *testing.T) {
```

Comment:
Did you succeed to run this test with an expected error? I mean without other code fixes.

I tried several times and they all passed without errors.

Reply:
I have managed to get it fail, but give that there is race condition that is hard to reproduce, you need to set numRepitions high and sometimes run several times.
I tried finding a way to simulate late arrival of events, but I didn't find a reasonable way to do so.

Reply:
I should add that my reasons for adding it was to at least see a flaky test if a race condition exists.